src/Components/Publishing/Layouts/ArticleWithFullScreen.tsx

```diff
@@ -90,3 +50,20 @@ export class ArticleWithFullScreen extends React.Component<
 }
 
 export default track()(ArticleWithFullScreen)
+
+export const getSlideshowImagesFromArticle = (article: ArticleData) => {
+  const fullscreenImages = []
+  let sectionIndex = 0
+
+  map(article.sections, section => {
+    if (includes(["image_collection", "image_set"], section.type)) {
+      map(section.images, image => {
+        image.setTitle = section.title
+        image.index = sectionIndex
```

Comment:
In general it's best to avoid mutation: 

```js
const img = cloneDeep(image)
// or if it's shallow 
const img = { 
  ...image, 
  setTitle: section.title
  index: sectionIndex
} 
```